### PR TITLE
Flat/framework

### PR DESCRIPTION
**Pull request by Flatfingers**
Numerous HUD, System Map, and general framework enhancements to support larger changes later.

### CODE REVIEW

File: script/GameObjects/Components/Core/Health.lua [hunk 1]
```diff
@@ -12,10 +12,26 @@ end
...
 function Entity:damage (amount, source)
   assert(self.health)
+
   if self.health <= 0 then return end
+
   self.health = max(0, self.health - amount)
   self:send(Event.Damaged(amount, source))
-  if self.health <= 0 then
+
+  if self.health <= 0.009999999 then
```
Comment (IllustrisJack) [inline script/GameObjects/Components/Core/Health.lua:21]: what is that lol :D

File: script/GameObjects/Components/Core/Health.lua [hunk 2]
```diff
@@ -59,6 +75,7 @@ function Entity:setHealth (value, max, rate)
 end
 
 function Entity:updateHealth (state)
-  if self:isDestroyed() then return end
-  self.health = min(self.healthMax, self.health + state.dt * self.healthRate)
+  if not self:isDestroyed() then
```
Comment (IllustrisJack) [inline script/GameObjects/Components/Core/Health.lua:78]: Personally i´d prefer the old one liner approach, opening an if-statement feels a bit unecessary

File: script/GameObjects/Components/Core/Health.lua (hunk 1 above)
Comment (Flatfingers) [reply script/GameObjects/Components/Core/Health.lua:21]: It's catching the case where health has been reduced to within two decimal places of zero. A test for exactly zero might not work as intended.

File: script/GameObjects/Components/Core/Health.lua (hunk 2 above)
Comment (Flatfingers) [reply script/GameObjects/Components/Core/Health.lua:78]: Future-proofing, that is. 😄 That area of the code is definitely going to change. When the game is close to stable and we do a maintenance cleanup pass through the code, one-liners are a good option.

File: script/GameObjects/Components/Core/Health.lua (hunk 1 above)
Comment (IllustrisJack) [reply script/GameObjects/Components/Core/Health.lua:21]: <= tests for below and exactly 0 tho, should be fine for most cases ;)

Comment (IllustrisJack) [reply script/GameObjects/Components/Core/Health.lua:21]: you can use math.floor to round health to 2 decimals If needed

Comment (Flatfingers) [reply script/GameObjects/Components/Core/Health.lua:21]: Code will be changed to use floor() in next commit.

**Review by IllustrisJack — APPROVED** [review]
Would really recommend to get some coding conventions written down since it´s a bit over the place but otherwise i´ts alright

**Review by dgavedissian — APPROVED** [review]

**Pull request merged**